src/src/com/microsoft/aad/adal/SerializedTokenCacheItem.java

```diff
@@ -0,0 +1,40 @@
+package com.microsoft.aad.adal;
+
+import com.google.gson.annotations.SerializedName;
+
+final class SerializedTokenCacheItem {
+
+    private static final String TAG = SerializedTokenCacheItem.class.getSimpleName();
+
+    @SerializedName("authority")
+    private final String mAuthority;
+
+    @SerializedName("refreshToken")
+    private final String mRefreshToken;
+
+    @SerializedName("idToken")
+    private final String mIdToken;
+
+    @SerializedName("familyClientId")
+    private final String mFamilyClientId;
+
+    SerializedTokenCacheItem(final TokenCacheItem tokenCacheItem) {
+        this.mAuthority = tokenCacheItem.getAuthority();
+        this.mRefreshToken = tokenCacheItem.getRefreshToken();
+        this.mIdToken = tokenCacheItem.getRawIdToken();
```

Comment:
And more over why don't you put UserInfo instead of IdToken. because this is what you've got from tokenCacheItem and what you put when you create TokenCacheItem. Why do you need to allocate this object in between when you could serialize UserInfo directly without this overhead